### PR TITLE
Close #12338: Remove goto from Tick128UpdateGuest

### LINKED ISSUE #12338: Remove goto from Tick128UpdateGuest
https://github.com/OpenRCT2/OpenRCT2/blob/d10b615eb33f6cd597340d83d68eaaa97c1a493e/src/openrct2/peep/Guest.cpp#L820-L823

### PR DESCRIPTION
**Pull request by frutiemax**

### CODE REVIEW

File: src/openrct2/peep/Guest.cpp [hunk 1]
```diff
@@ -820,7 +995,7 @@ void Guest::Tick128UpdateGuest(int32_t index)
                     HappinessTarget = std::max(HappinessTarget - 128, 0);
                     peep_leave_park(this);
                     peep_update_hunger(this);
-                    goto loc_68F9F3;
+                    UpdateStatistics();
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Guest.cpp:998]: Must return after this statement

File: src/openrct2/peep/Guest.cpp [hunk 2]
```diff
@@ -694,6 +694,181 @@ int32_t Guest::CheckEasterEggName(int32_t index) const
...
+            {
+                Action = PEEP_ACTION_THROW_UP;
+                ActionFrame = 0;
+                ActionSpriteImageOffset = 0;
+                UpdateCurrentActionSpriteType();
+            }
+        }
+    }
+
+    // Remaining content is executed every call.
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Guest.cpp:741]: You have changed the logic here. This needs to be split up into two functions as this second part from here is called every time but the first part is only called sometimes. That's what this comment is for.

File: src/openrct2/peep/Guest.cpp [hunk 3]
```diff
@@ -694,6 +694,181 @@ int32_t Guest::CheckEasterEggName(int32_t index) const
...
+    if (newNausea != Nausea)
+    {
+        Nausea = newNausea;
+        WindowInvalidateFlags |= PEEP_INVALIDATE_PEEP_2;
+    }
+}
+
 void Guest::Tick128UpdateGuest(int32_t index)
 {
     if (static_cast<uint32_t>(index & 0x1FF) == (gCurrentTicks & 0x1FF))
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Guest.cpp:874]: Function should be split up into two here.

File: src/openrct2/peep/Guest.cpp [hunk 4]
```diff
@@ -694,6 +694,181 @@ int32_t Guest::CheckEasterEggName(int32_t index) const
     return _stricmp(buffer, gPeepEasterEggNames[index]) == 0;
 }
 
+void Guest::UpdateStatistics()
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Guest.cpp:697]: These aren't really statistics can you come up with a better name

File: src/openrct2/peep/Guest.cpp (hunk 2 above)
Comment (frutiemax) [reply src/openrct2/peep/Guest.cpp]: You are right, I've pretty much reverted all my changes. I haven't found a proper name for these functions.

**Comment by duncanspumpkin** [comment]
Its not great the new names but at least its easy to follow and someone can improve the names later. This refactor also helped @ZehMatt and myself realise there is a desync issue with the entity lists.

**Pull request merged**